specs/disk_details_for_hardware_inventory.adoc

```diff
@@ -0,0 +1,156 @@
+= Specification for adding disk details in node hardware inventory
+
+Add disk details also as a part of node hardware inventory
+
+== Problem description
+
+Hardware inventory details are used to show exact status of the particular
+node. Currently disk details are missing in node hardware inventory. Add disk
+details in node hardware inventory and disk details should contain device_name,
+fs_type, fs_uuid, model, mount_point, parent, size, name, disk_type, used, ssd,
+vendor, disk_id.
+
+== Use Cases
+
+Tendrl should show the current status of disk.
+
+== Proposed change
+
+* "Hdparm" is a package used to find the disk serial number and  firmware
+  version.
+* Add Hdparm as dependency.
+* From node_agent add a new function in pull_hardware_inventory to find disk
+  details and update those details into node_inventory list.
+* Modify the manager to retrieve inventory details from tendrl-node-inventory
+  file and compare it with node_inventory list, If any changes then dump the
+  disk details in etcd.
+* Store free disk details with different key and used disk details with
+  different key.
+* Create objects in tendrl definition for used disks and free disks.
+
+== Alternatives
+
+None
+
+== Data model impact
+
+* Add used disk details in etcd with key nodes/node_id/Used_disks.
+* Add free disk details in etcd with key nodes/node_id/Free_disks.
+* Add two different object in tendrl definitions for used_disk and free_disk.
+```
+     list: nodes/$Node_context.node_id/Free_disks
+     list: nodes/$Node_context.node_id/Used_disks
+```
+
+== Impacted modules
+
+== Tendrl API impact:
+
+None
+
+=== Tendrl/common impact
+
+None
+
+=== Tendrl/node_agent impact
+
+* Add a new function in pull_hardware_inventory to collect disk details.
+* Modify the manager to retrieve inventory details from tendrl-node-inventory
+  file and compare it with node_inventory list, If any changes then dump the
+  disk details in etcd.
+* Store free disk details with different key and used disk details with
+  different key.
+* Create objects in tendrl definition for used disks and free disks.
+* Commands used to retrieve disk details are:
+```
+1. columns = 'NAME,KNAME,FSTYPE,MOUNTPOINT,UUID,PARTUUID,MODEL,SIZE,TYPE,PKNAME,
+   VENDOR'
+   lsblk --all --bytes --noheadings --output='%s' --path --raw" % columns
+2. hdparm -I disk_name
```

Comment:
@nthomas-redhat it is not come by default , we have to specify. I have added hdparm in dependency.